ecmascript/transforms/classes/src/super_field.rs

```diff
@@ -101,7 +82,120 @@ impl<'a> Fold for SuperCalleeFolder<'a> {
             _ => {}
         }
 
-        let n = match n {
+        // We pretend method folding mode for while folding injected `_defineProperty`
+        // calls.
+        match n {
+            Expr::Call(CallExpr {
+                callee: ExprOrSuper::Expr(ref expr),
+                ..
+            }) => match &**expr {
+                Expr::Ident(Ident {
+                    sym: js_word!("_defineProperty"),
+                    ..
+                }) => {
+                    let old = self.in_injected_define_property_call;
+                    self.in_injected_define_property_call = true;
+                    let n = n.fold_children_with(self);
+                    self.in_injected_define_property_call = old;
+                    return n;
+                }
+                _ => {}
+            },
+            _ => {}
+        }
+
+        let n = self.fold_super_member_call(n);
+        let n = self.fold_super_member_set(n);
+        let n = self.fold_super_member_get(n);
+
+        n.fold_children_with(self)
+    }
+}
+
+impl<'a> SuperFieldAccessFolder<'a> {
+    /// # In
+    /// ```js
+    /// super.foo(a)
+    /// ```
+    /// # out
+    /// ```js
+    /// _get(_getPrototypeOf(Clazz.prototype), 'foo', this).call(this, a)
+    /// ```
+    fn fold_super_member_call(&mut self, n: Expr) -> Expr {
+        match n {
+            Expr::Call(CallExpr {
+                callee: ExprOrSuper::Expr(ref callee_expr),
+                ref args,
+                ref type_args,
+                ..
+            }) => match &**callee_expr {
+                Expr::Member(MemberExpr {
+                    obj:
+                        ExprOrSuper::Super(Super {
+                            span: super_token, ..
+                        }),
+                    prop,
+                    computed,
+                    ..
+                }) => {
+                    let this = match self.constructor_this_mark {
+                        Some(mark) => quote_ident!(DUMMY_SP.apply_mark(mark), "_this").as_arg(),
+                        _ => ThisExpr { span: DUMMY_SP }.as_arg(),
+                    };
+
+                    let callee = self.super_to_get_call(*super_token, prop.clone(), *computed);
```

Comment:
I don't know how to prevent `clone` in this match branch. I try to use *prop, but the compiler does not allow to do this.

Reply:
You can remove clone by changing `callee: ExprOrSuper::Expr(ref callee_expr), ` to `callee: ExprOrSuper::Expr(callee_expr)`, but I don't think it worth.
It requires reconstructing when `callee_expr` is not a member expression we wanted, and it means dirty code, and `memmove`.

I think clone is ok in this case, as `expr` in super[expr]` is very simple in most cases.